python/heterocl/compute_api.py

```diff
@@ -921,5 +921,57 @@ def reduce_body():
     make_reduce.__doc__ = doc_str.format(name)
     return make_reduce
 
+
+def bitcast(tensor, dst_dtype, name=None):
+    """Bitcast a HeteroCL tensor to the destination data type of the same bitwidth.
+
+    This API **bitcast** the input tensor from its own data type (source dtype)
+    to the destination data type (dst_dtype). The destination data type must have
+    the same bitwidth with the source datatype. 
+
+    Parameters
+    ----------
+    tensor: Tensor or Expr
+        The input tensor or expression of the source data type
+    dst_dtype: Type
+        The destination data type. For example, hcl.UInt(32)
+    Name: str
+        The name of the returned tensor
+    
+    Returns
+    -------
+    _tensor: Tensor
+        The bitcasted tensor of destination data ype
+    """
+    
+    # check type
+    if not isinstance(tensor, Tensor) and not isinstance(tensor, _expr.Expr):
+        raise APIError("bitcast input tensor must be HeteroCL Tensor or expression.")
+
+    if not isinstance(dst_dtype, Type):
+        raise APIError("dst_dtype should be HeteroCL data type.")
+
+    # check bitwidth
+    src_bitwidth = get_type(tensor.dtype)[1]
+    dst_bitwidth = dst_dtype.bits 
+    if src_bitwidth != dst_bitwidth:
+        if isinstance(tensor, Tensor):
+            raise APIError(f"Bitcasting source Tensor of bitwidth {src_bitwidth} to destination bitwidth {dst_bitwidth}.") 
+        else:
+            raise APIError(f"Bitcasting source expression of bitwidth {src_bitwidth} to destination bitwidth {dst_bitwidth}.")
```

Comment:
The error message does not tell the right thing. We should at least say the bitwidths are mismatched. Then, we can provide additional info.

Reply:
Rewrote the error message